app/lib/ui/views/active/form_field/address_form_field.dart

```diff
@@ -1,88 +1,99 @@
   // https://pub.dev/packages/country_state_city_picker
-  TextEditingController _streetController;
-  TextEditingController _cityController;
-  TextEditingController _stateController;
-  TextEditingController _countryController;
-  TextEditingController _postCodeController;
-
-  Address getAddress() => Address(
-      streetAddress: _streetController.text,
-      city: _cityController.text,
-      provinceOrState: _stateController.text,
-      country: _countryController.text,
-      postalCode: _postCodeController.text);
+  String _street;
+  String _city;
+  String _provinceOrState;
+  String _country;
+  String _postalCode;
 
   @override
   void initState() {
-    _streetController =
-        TextEditingController(text: widget.initialAddr.streetAddress);
-    _cityController = TextEditingController(text: widget.initialAddr.city);
-    _stateController =
-        TextEditingController(text: widget.initialAddr.provinceOrState);
-    _countryController =
-        TextEditingController(text: widget.initialAddr.country);
-    _postCodeController =
-        TextEditingController(text: widget.initialAddr.postalCode);
+    _street = widget.initialAddr.streetAddress;
+    _city = widget.initialAddr.city;
+    _provinceOrState = widget.initialAddr.provinceOrState;
+    _country = widget.initialAddr.country;
+    _postalCode = widget.initialAddr.postalCode;
     super.initState();
   }
 
-  @override
-  void dispose() {
-    _streetController.dispose();
-    _cityController.dispose();
-    _stateController.dispose();
-    _countryController.dispose();
-    _postCodeController.dispose();
-    super.dispose();
+  void _saveAll() {
+    widget.onSaved(Address(
+        streetAddress: _street,
+        city: _city,
+        provinceOrState: _provinceOrState,
+        country: _country,
+        postalCode: _postalCode));
   }
 
   @override
   Widget build(BuildContext context) {
-    return Column(children: [
-      ListTile(
-          title: Text("Street"),
-          subtitle: TextFormField(
-            controller: _streetController,
-          )),
-      ListTile(
-          title: Text("City"),
-          subtitle: TextFormField(
-            controller: _cityController,
-          )),
-      ListTile(
-          title: Text("Province or State"),
-          subtitle: TextFormField(
-            controller: _stateController,
-          )),
-      ListTile(
-          title: Text("Country"),
-          subtitle: TextFormField(
-            controller: _countryController,
-          )),
-      ListTile(
-          title: Text("Postal Code"),
-          subtitle: TextFormField(
-            controller: _postCodeController,
-          )),
-    ]);
+    return Form(
+      key: widget.formKey,
+      child: Column(children: [
+        StringFormField(
+            initial: _street,
+            title: "Street",
+            onSaved: (String changed) {
+              _street = changed;
+              _saveAll();
```

Comment:
You may be thinking "Woah, that's a lotta saves!" but these saves go to temporary objects that, only once the person clicks submit or exits the editing screen do the changes get communicated to the database.